compilers/concrete-optimizer/concrete-optimizer/src/optimization/config.rs

```diff
@@ -29,12 +37,21 @@ pub struct SearchSpace {
 }
 
 impl SearchSpace {
-    pub fn default_cpu() -> Self {
-        let glwe_log_polynomial_sizes: Vec<u64> = DEFAUT_DOMAINS
-            .glwe_pbs_constrained
-            .log2_polynomial_size
-            .as_vec();
-        let glwe_dimensions: Vec<u64> = DEFAUT_DOMAINS.glwe_pbs_constrained.glwe_dimension.as_vec();
+    pub fn default_cpu(parameter_restrictions: ParameterRestrictions) -> Self {
+        let mut glwe_log_polynomial_sizes_range = DEFAUT_DOMAINS.glwe_pbs_constrained.log2_polynomial_size;
+        glwe_log_polynomial_sizes_range.start = std::cmp::max(glwe_log_polynomial_sizes_range.start,
+            parameter_restrictions.log2_polynomial_size_min);
+        glwe_log_polynomial_sizes_range.end = std::cmp::min(glwe_log_polynomial_sizes_range.end,
+            parameter_restrictions.log2_polynomial_size_max);
+        let glwe_log_polynomial_sizes: Vec<u64> = glwe_log_polynomial_sizes_range.as_vec();
+
+        let mut glwe_dimensions_range = DEFAUT_DOMAINS.glwe_pbs_constrained.glwe_dimension;
+        glwe_dimensions_range.start = std::cmp::max(glwe_dimensions_range.start,
+            parameter_restrictions.glwe_dimension_min);
+            glwe_dimensions_range.end = std::cmp::min(glwe_dimensions_range.end,
+            parameter_restrictions.glwe_dimension_max);
+        let glwe_dimensions: Vec<u64> = glwe_dimensions_range.as_vec();
+
```

Comment:
From what I understand, the use of `std::cmp::min` and `std::cmp::max` is a way to softly encode an `std::optional`. I think an explicit one would be better here.

Reply:
That's also what I understand, but the thing that bother me is that it restrict the search space to the intersection of the search space of our cpu backend and the parameter_restriction, it would be better to let all the freedom to define the search space as mainly the restriction of the parameter space is due of implementation details of tfhe-rs (if well remember only the polynomial size supported in our fft) and hardware provider may have a larger search space.
To resume I think I would prefer a way to define a custom search space instead of only a restrictions of the cpu one.